### PR TITLE
Convert sac cart pole: gin -> python config

### PR DESCRIPTION
**Pull request by breakds**
# Motivation

In my recent quest on retiring one time experience replayer, I need an example of true off policy training (i.e. actually sample from replay buffer). I picked SAC cart pole and converted it to python conf as we are retiring `gin` as well.

# Solution

1. Converted gin -> py conf
2. Updated the unit test

# Testing
CI passed. Also tested the training locally and get

sac_cart_pole

Which resembles the current graph under examples.

### CODE REVIEW

File: alf/examples/sac_cart_pole_conf.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+alf.config(
+    'TrainerConfig',
+    algorithm_ctor=SacAlgorithm,
+    initial_collect_steps=1000,
+    mini_batch_length=2,
+    mini_batch_size=64,
+    unroll_length=1,
+    num_updates_per_train_iter=1,
+    whole_replay_buffer_training=False,
+    clear_replay_buffer=False,
```
Comment (hnyu) [inline alf/examples/sac_cart_pole_conf.py]: Since you've imported sac_conf.py, you don't have to re-configure some options in this conf file.

**Review by hnyu — CHANGES_REQUESTED** [review]

File: alf/examples/sac_cart_pole_conf.py (hunk 1 above)
Comment (breakds) [reply alf/examples/sac_cart_pole_conf.py]: Ah yep. Thanks for pointing out. Fixed.

**Comment by breakds** [comment]
Updated. PTAL.

File: alf/examples/sac_cart_pole_conf.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# environment config
+alf.config(
+    'create_environment', env_name="CartPole-v0", num_parallel_environments=8)
+
+# algorithm config
+alf.config('ActorDistributionNetwork', fc_layer_params=(100, ))
+alf.config('QNetwork', fc_layer_params=(100, ))
+alf.config(
+    'SacAlgorithm',
+    actor_network_cls=ActorDistributionNetwork,
```
Comment (hnyu) [inline alf/examples/sac_cart_pole_conf.py]: Could skip actor_network_cls for discrete actions

Comment (breakds) [reply alf/examples/sac_cart_pole_conf.py]: Thanks for pointing out. Updated.

**Review by hnyu — APPROVED** [review]

**Pull request merged**